### PR TITLE
fix(php): emit required global headers as named constructor arguments

### PR DESCRIPTION
**Pull request by lifanzou**
## Summary
- Fix PHP dynamic snippet generator to emit required global headers as named constructor arguments instead of inside the `options` array
- Split `getConstructorHeaderArgs` into `getRequiredGlobalHeaderArgs` (positional) and `getOptionalGlobalHeaderArgs` (options array), using a dedicated `isHeaderTypeOptional` predicate that correctly handles nullable alias types
- Adds `generatePlaceholderValueForRequiredHeader` for fallback values when snippet data is missing

## Context
When the OpenAPI importer turns an `apiKey-in-header` security scheme into a required global header, the PHP SDK constructor takes that header as a required positional parameter (e.g. `string $authorization`). The snippet generator was placing these in the `options` array, producing constructor calls that omitted the required parameter — causing phpstan failures.

## Test plan
- [x] All 32 existing PHP dynamic snippet tests pass
- [x] Snapshot updated: `tenantId` moves from `options` array to named constructor arg
- [x] Adversarial review (2 rounds) — identified and fixed nullable-alias edge case
- [x] Independent cold review approved

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: generators/php/dynamic-snippets/src/__test__/__snapshots__/DynamicSnippetsGenerator.test.ts.snap [hunk 1]
```diff
@@ -635,10 +635,8 @@ namespace Example;
 use Acme\\AcmeClient;
 
 $client = new AcmeClient(
+    tenantId: 'my-tenant-id',
```
Comment (iamnamananand996) [inline generators/php/dynamic-snippets/src/__test__/__snapshots__/DynamicSnippetsGenerator.test.ts.snap:638]: look good

**Review by iamnamananand996 — APPROVED** [review]
